### PR TITLE
Ptx 23576 VolResizeAllVolumes in parallel

### PR DESCRIPTION
**Pull request by vprabhakar-px**
INFO[2024-04-28 11:41:14] Verifying : Description : Test completed successfully ? 
INFO[2024-04-28 11:41:14] Actual:PASS, Expected: PASS         
INFO[2024-04-28 11:41:17] --------Test End------                       
INFO[2024-04-28 11:41:17] #Test: VolResizeAllVolumes                   
INFO[2024-04-28 11:41:17] #Description: Trigger vol resize on all volumes at once  
INFO[2024-04-28 11:41:17] ------------------------                     
• [1014.778 seconds]
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[AfterSuite] 
...
INFO[2024-04-28 11:42:05] Verifying : Description : Validate cleanup operation successful? 
INFO[2024-04-28 11:42:05] Actual:true, Expected: true                  
INFO[2024-04-28 11:42:06] Verifying : Description : Test completed successfully ? 
INFO[2024-04-28 11:42:06] Actual:PASS, Expected: PASS         
INFO[2024-04-28 11:42:07] --------Test End------                       
INFO[2024-04-28 11:42:07] #Test: Validate Cleanup                      
INFO[2024-04-28 11:42:07] #Description: Validating clean up            
INFO[2024-04-28 11:42:07] ------------------------                     
INFO[2024-04-28 11:42:08] Verifying : Description : Test completed successfully ? 
INFO[2024-04-28 11:42:08] Actual:PASS, Expected: PASS         
INFO[2024-04-28 11:42:09] --------Test End------                       
INFO[2024-04-28 11:42:09] #Test: Validate Cleanup                      
INFO[2024-04-28 11:42:09] #Description: Validating clean up            
INFO[2024-04-28 11:42:09] ------------------------                     
INFO[2024-04-28 11:42:10] Dashboard URL : https://aetos.pwx.purestorage.com/resultSet/testSetID/627299 
[AfterSuite] PASSED [52.300 seconds]
------------------------------

Ran 1 of 340 Specs in 1297.681 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 339 Skipped
PASS

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -1503,10 +1503,10 @@ func (d *portworx) ValidateCreateVolume(volumeName string, params map[string]str
 				return errFailedToInspectVolume(volumeName, k, requestedSpec.AggregationLevel, vol.Spec.AggregationLevel)
 			}
 			/* Ignore shared setting.
-		case api.SpecShared:
-			if requestedSpec.Shared != vol.Spec.Shared {
-				return errFailedToInspectVolume(volumeName, k, requestedSpec.Shared, vol.Spec.Shared)
-			}
+			case api.SpecShared:
```
Comment (dbhatnagar-px) [inline drivers/volume/portworx/portworx.go:1506]: Is this indentation correct ?

File: tests/basic/storage_pool_test.go [hunk 2]
```diff
@@ -2,6 +2,7 @@ package tests
 
 import (
 	"fmt"
+	corev1 "k8s.io/api/core/v1"
```
Comment (dbhatnagar-px) [inline tests/basic/storage_pool_test.go:5]: Please keep the imports clean in sections

File: drivers/volume/portworx/portworx.go (hunk 1 above)
Comment (vprabhakar-px) [reply drivers/volume/portworx/portworx.go:1506]: yes, this is correct , its aligned right as the code was commented

File: tests/basic/storage_pool_test.go (hunk 2 above)
Comment (vprabhakar-px) [reply tests/basic/storage_pool_test.go:5]: again modified by goland

File: tests/basic/storage_pool_test.go [hunk 3]
```diff
@@ -12080,3 +12081,234 @@ var _ = Describe("{PoolDeleteMultiplePools}", func() {
...
+			log.Infof("Resizing Volume [%v] from size [%v] to [%v]", newVolumeIDs.vol.Name, newVolumeIDs.vol.Size, volNewSize*units.GiB)
+			vol, err := Inst().S.ResizePVC(newVolumeIDs.ctx, newVolumeIDs.pvc, volNewSize)
+			log.FailOnError(err, "Failed to resize PVC [%v]", vol.Name)
+
+			time.Sleep(60 * time.Second)
+			volReSize, err := Inst().V.InspectVolume(vol.ID)
+			log.FailOnError(err, "inspect returned error ?")
+
+			log.Infof("Verify volume size after resizing the volume [%v]", vol.Name)
+			dash.VerifyFatal(volReSize.Spec.Size > newVolumeIDs.vol.Size, true, "Resize of Volume didnot happen!")
```
Comment (dbhatnagar-px) [inline tests/basic/storage_pool_test.go:12187]: Can we also add check for the new size apart from simply checking less than and greater than.

File: tests/basic/storage_pool_test.go [hunk 4]
```diff
@@ -12080,3 +12081,234 @@ var _ = Describe("{PoolDeleteMultiplePools}", func() {
...
+
+			// Resize Volume and verify if volume resized
+			log.FailOnError(Inst().V.ResizeVolume(volId, toResize), "Failed to resize volume")
+
+			//Size after resize
+			sizeAfterResize, err := Inst().V.InspectVolume(volId)
+			log.FailOnError(err, "inspect returned error ?")
+
+			dash.VerifyFatal(sizeBeforeResize.Spec.Size < sizeAfterResize.Spec.Size, true, "Volume resize did not happen")
+
```
Comment (dbhatnagar-px) [inline tests/basic/storage_pool_test.go:12206]: Same comment as above - let's also add check for the newly resized size apart from less than or greater than previous size

File: tests/common.go [hunk 5]
```diff
@@ -12129,3 +12136,12 @@ func RefreshIscsiSession(n node.Node) error {
 	}
 	return nil
 }
+
+// GetPVCObjFromVol Returns pvc object from Volume
+func GetPVCObjFromVol(vol *volume.Volume) (*v1.PersistentVolumeClaim, error) {
+	pvcObj, err := k8sCore.GetPersistentVolumeClaim(vol.Name, vol.Namespace)
+	if err != nil {
```
Comment (dbhatnagar-px) [inline tests/common.go:12143]: Minor comment - we can simply return pvcObj, err here as you are anyways doing log.FailOnError in the test case so this condition is being checked there

Comment (vprabhakar-px) [reply tests/common.go:12143]: Done

File: tests/basic/storage_pool_test.go (hunk 4 above)
Comment (vprabhakar-px) [reply tests/basic/storage_pool_test.go:12221]: done

**Review by dbhatnagar-px — APPROVED** [review]
Lgtm

**Pull request merged**